ui/selectors/assets.ts

```diff
@@ -210,3 +211,65 @@ export const getMultiChainAssets = createDeepEqualSelector(
     });
   },
 );
+
+export const getMultichainAggregatedBalance = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainNetwork,
+  getMultichainBalances,
+  getAccountAssets,
+  getAssetsRates,
+  (
+    selectedAccountAddress,
+    currentNetwork,
+    multichainBalances,
+    accountAssets,
+    assetRates,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+
+    let aggregatedBalance = new BigNumber(0);
+
+    assetIds.forEach((assetId: CaipAssetId) => {
+      const { chainId } = parseCaipAssetType(assetId);
+      if (chainId === currentNetwork.chainId) {
+        const balance = balances?.[assetId] || { amount: '0', unit: '' };
+        const rate = assetRates?.[assetId]?.rate || '0';
+        const balanceInFiat = new BigNumber(balance.amount).times(rate);
+
+        aggregatedBalance = aggregatedBalance.plus(balanceInFiat);
+      }
+    });
+
+    return aggregatedBalance.toNumber();
+  },
+);
+
+export const getMultichainNativeTokenBalance = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainNetwork,
+  getMultichainBalances,
+  getAccountAssets,
+  (
+    selectedAccountAddress,
+    currentNetwork,
+    multichainBalances,
+    accountAssets,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+
+    let nativeTokenBalance = { amount: 0, unit: '' };
+
+    assetIds.forEach((assetId: CaipAssetId) => {
+      const { chainId, assetNamespace } = parseCaipAssetType(assetId);
+      if (chainId === currentNetwork.chainId && assetNamespace === 'slip44') {
+        const balance = balances?.[assetId] || { amount: 0, unit: '' };
```

Comment:
The fallback should be `nativeTokenBalance`, right?

Reply:
Hmmm, in which scenario would this be the case? If an `assetId` exists in a wallet, but the `balance` doesn't, to me it makes more sense to fallback to a `0` balance instead.

Reply:
Oh, I see what you mean. Maybe we should rename `nativeTokenBalance` to be `zeroBalanceFallback` so that it can be reused without confusion